packages/@aws-cdk/aws-lambda/test/function.test.ts

```diff
@@ -777,29 +776,9 @@ describe('function', () => {
       tracing: lambda.Tracing.DISABLED,
     });
 
-    expect(stack).not.toHaveResource('AWS::IAM::Policy', {
-      PolicyDocument: {
-        Statement: [
-          {
-            Action: [
-              'xray:PutTraceSegments',
-              'xray:PutTelemetryRecords',
-            ],
-            Effect: 'Allow',
-            Resource: '*',
-          },
-        ],
-        Version: '2012-10-17',
-      },
-      PolicyName: 'MyLambdaServiceRoleDefaultPolicy5BBC6F68',
-      Roles: [
-        {
-          Ref: 'MyLambdaServiceRole4539ECB6',
-        },
-      ],
-    });
+    Template.fromStack(stack).resourceCountIs('AWS::IAM::Policy', 0);
```

Comment:
This is not strictly equivalent. If this generates a new policy tomorrow, the new test might fail while the other one is still alright. 

I guess it might be fine, but feels risky.